arithmetization/src/main/java/net/consensys/linea/zktracer/module/limits/precompiles/BlakeRounds.java

```diff
@@ -15,18 +15,66 @@
 
 package net.consensys.linea.zktracer.module.limits.precompiles;
 
+import static java.lang.Integer.MAX_VALUE;
+
+import java.math.BigInteger;
+
+import com.google.common.base.Preconditions;
 import lombok.Getter;
+import lombok.Setter;
 import lombok.experimental.Accessors;
 import net.consensys.linea.zktracer.container.module.CountingOnlyModule;
 import net.consensys.linea.zktracer.container.stacked.CountOnlyOperation;
+import org.apache.tuweni.bytes.Bytes;
 
 @Getter
 @Accessors(fluent = true)
 public final class BlakeRounds implements CountingOnlyModule {
   private final CountOnlyOperation counts = new CountOnlyOperation();
+  @Setter private boolean transactionBundleContainsIllegalOperation = false;
+
+  private static final BigInteger INTEGER_MAX_VALUE_BI = BigInteger.valueOf(MAX_VALUE);
 
   @Override
   public String moduleKey() {
     return "PRECOMPILE_BLAKE_ROUNDS";
   }
+
+  @Override
+  public void addPrecompileLimit(final int count) {
+    throw new UnsupportedOperationException("Not implemented");
+  }
+
+  public void addPrecompileLimit(final Bytes r) {
+    Preconditions.checkArgument(r.bitLength() <= 32, "r is at most 4 bytes long");
```

Comment:
I wonder whether it would make more sense to constrain it to
```java
r.size() <= 4
```
as `r` ought to be extracted as the first 4 bytes of call data, and `r.bitlength()` ignores the first bytes if they are zero.

Reply:
We may even want to assert `r.size() == 4`.